blocklylib-core/src/main/java/com/google/blockly/android/ui/TrashCanView.java

```diff
@@ -0,0 +1,196 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.blockly.android.ui;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.drawable.Drawable;
+import android.support.annotation.IntDef;
+import android.support.v4.content.ContextCompat;
+import android.util.AttributeSet;
+import android.util.Log;
+import android.view.DragEvent;
+import android.view.View;
+import android.widget.ImageView;
+
+import com.google.blockly.android.AbstractBlocklyActivity;
+import com.google.blockly.android.R;
+import com.google.blockly.android.control.BlocklyController;
+
+import java.lang.annotation.Retention;
+
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+/**
+ * Drop target view for deleting blocks via a drag gesture, with animation.
+ * <p/>
+ * This view has two layout attributes, {@code closedIcon} and {@code openedIcon}. Each is a
+ * reference to a drawable resource for one of the two trash states, closed (default/idle state) and
+ * opened (pending drop during drag).  The TrashCanView assumes both drawables are the same size, so
+ * the overall view will not change size when the state changes.
+ */
+public class TrashCanView extends ImageView {
+    private static final String TAG = "TrashCanView";
+
+    @Retention(SOURCE)
+    @IntDef({STATE_DEFAULT, STATE_ON_HOVER})
+    public @interface HoverState {}
+    private static final int STATE_DEFAULT = 0;
+    private static final int STATE_ON_HOVER = 1;
+
+    protected int mState = STATE_DEFAULT;
+    protected Drawable mDefaultDrawable;
+    protected Drawable mOnHoverDrawable;
+
+    public TrashCanView(Context context) {
+        this(context, null, 0);
+    }
+
+    public TrashCanView(Context context, AttributeSet attrs) {
+        this(context, attrs, 0);
+    }
+
+    public TrashCanView(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        buildUI();
+
+        TypedArray a = context.getTheme().obtainStyledAttributes(
+                attrs,
+                R.styleable.TrashCanView,
+                0, 0);
+        try {
+            //noinspection ResourceType
+            setDefaultIcon(a.getResourceId(
+                    R.styleable.TrashCanView_defaultIcon, R.drawable.blockly_trash));
+            setOnHoverIcon(a.getResourceId(
+                    R.styleable.TrashCanView_onHoverIcon, R.drawable.blockly_trash_open));
+        } finally {
+            a.recycle();
+        }
+    }
+
+    /**
+     * Assigns the active {@link BlocklyController} for this TrashView.
+     *
+     * @param controller
+     */
+    public void setBlocklyController(BlocklyController controller) {
```

Comment:
Not sure if I like your current implementation more or less, but here's another approach to consider. If you like it as is feel free to submit.

Instead of posting a runnable and having the buildUI() method the AbstractBlocklyActivity or the Controller could create and set the OnDragToTrashListener. This would remove the direct dependencies on BlocklyController and AbstractBlocklyActivity from this view at the cost of making it less clear what happens when a block is dragged over the trash.